### PR TITLE
[SofaExporter] Rename OBJExporter into VisualModelOBJExporter

### LINKED ISSUE #1783: Rename OBJExporter to VisualExporter
"OBJExporter" implies that the component exports any thing with positions and a topology to OBJ files.
But actually OBJExporter exports all the Visual Models from its context.
(so if it is at the root, it will export all the visual models into a single OBJ file)

So the goal would be to rename OBJExporter  to VisualExporter (or any meaningful name)

(was introduced in issue #13)

### PR DESCRIPTION
**Pull request by hugtalbot**
Rename OBJExporter into VisualModelOBJExporter to express more clearly that it only exports the VisualModels
Compatibility layer added + Alias.

Fix #1783

Q: does it make sense to keep examples in the module itself?

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by guparan** [comment]
I cleaned the VisualModelOBJExporter alias, using the SOFA_DEPRECATED_ATTRIBUTE macro.
Doing that, I also updated the status of SOFAEXPORTER_NAMESPACE deprecation (from deprecated to disabled).

File: SofaKernel/modules/SofaSimulationCore/src/sofa/simulation/ExportVisualModelOBJVisitor.h [hunk 1]
```diff
@@ -60,8 +58,9 @@ class SOFA_SIMULATION_CORE_API ExportOBJVisitor : public Visitor
     int count;
 };
 
+using ExportOBJVisitor [[deprecated("The ExportOBJVisitor is now deprecated, please use ExportVisualModelOBJVisitor instead. Compatibility stops at v22.06")]] = ExportVisualModelOBJVisitor;
```
Comment (guparan) [inline SofaKernel/modules/SofaSimulationCore/src/sofa/simulation/ExportVisualModelOBJVisitor.h]: TODO: use SOFA_ATTRIBUTE_DEPRECATED macro here too.

**Review by guparan — CHANGES_REQUESTED** [review]

**Comment by guparan** [comment]
[ci-build][with-all-tests]

**Review by guparan — CHANGES_REQUESTED** [review]
TODO: add backward compat headers to forward `#include <SofaExporter/OBJExporter.h>` and `#include <SofaSimulationCore/ExportOBJVisitor.h>`

File: SofaKernel/modules/Sofa.Compat/src/SofaSimulationCore/OBJExporter.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <SofaSimulationCore/VisualModelOBJExporter.h>
+
+// SOFA_DEPRECATED_HEADER("v21.12", "v22.06", "SofaSimulationCore/OBJExporter.h")
+
```
Comment (fredroy) [inline SofaKernel/modules/Sofa.Compat/src/SofaSimulationCore/OBJExporter.h]:
```suggestion
SOFA_DEPRECATED_HEADER("v21.12", "v22.06", "SofaSimulationCore/OBJExporter.h")
```

**Review by fredroy — CHANGES_REQUESTED** [review]

File: modules/SofaExporter/src/SofaExporter/VisualModelOBJExporter.h [hunk 3]
```diff
@@ -29,44 +29,47 @@
...
-    = _objexporter_::OBJExporter;
+    SOFA_ATTRIBUTE_DISABLED__SOFAEXPORTER_NAMESPACE_2106()
+    = DeprecatedAndRemoved;
 namespace misc {
     using OBJExporter
-        SOFA_ATTRIBUTE_DEPRECATED__SOFAEXPORTER_NAMESPACE_1712()
-        = _objexporter_::OBJExporter;
+        SOFA_ATTRIBUTE_DISABLED__SOFAEXPORTER_NAMESPACE_1712()
+        = DeprecatedAndRemoved;
 } // namespace misc
```
Comment (guparan) [inline modules/SofaExporter/src/SofaExporter/VisualModelOBJExporter.h]: To be moved to `Sofa.Compat/src/SofaSimulationCore/OBJExporter.h`

File: modules/SofaExporter/src/SofaExporter/VisualModelOBJExporter.h [hunk 4]
```diff
@@ -29,44 +29,47 @@
...
+} // namespace _visualmodelobjexporter_
 
 // Import the object in the exporter namespace to avoid having all the object straight in component.
-namespace exporter {
-    using OBJExporter = _objexporter_::OBJExporter;
+namespace exporter
+{
+    using OBJExporter
+        SOFA_ATTRIBUTE_DEPRECATED("v21.12", "v22.06", "Use VisualModelOBJExporter instead.")
+        = _visualmodelobjexporter_::VisualModelOBJExporter;
```
Comment (guparan) [inline modules/SofaExporter/src/SofaExporter/VisualModelOBJExporter.h]: To be moved to `Sofa.Compat/src/SofaSimulationCore/OBJExporter.h`

**Review by guparan — CHANGES_REQUESTED** [review]

**Comment by guparan** [comment]
I removed the [ci-depends] as this PR must provide a compat layer (already done actually) not to break SofaPython3.

**Comment by guparan** [comment]
[ci-build][with-all-tests]

**Review by guparan — APPROVED** [review]

**Pull request merged**